document-store/src/main/java/org/hypertrace/core/documentstore/mongo/parser/MongoProjectionSelectTypeExpressionParser.java

```diff
@@ -10,14 +10,15 @@
 import org.hypertrace.core.documentstore.expression.impl.ConstantExpression;
 import org.hypertrace.core.documentstore.expression.impl.FunctionExpression;
 import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
-import org.hypertrace.core.documentstore.expression.type.SelectingExpression;
+import org.hypertrace.core.documentstore.expression.type.SelectTypeExpression;
 
 @Slf4j
-public final class MongoProjectionSelectingExpressionParser extends MongoSelectingExpressionParser {
+public final class MongoProjectionSelectTypeExpressionParser
```

Comment:
I would still keep it `MongoProjectionSelectingExpressionParser` or simply `MongoProjectingExpressionParser` or `MongoProjectingSelectionParser`. 
I don't feel `MongoProjectionSelectTypeExpressionParser` improves readability.

Reply:
I would prefer `MongoProjectingExpressionParser`